### PR TITLE
Added Linear detrend function for data that has more than one dimension

### PR DESCRIPTION
**Pull request by roxyboy**

### CODE REVIEW

File: xrft/xrft.py [hunk 1]
```diff
@@ -38,7 +41,39 @@ def _hanning(da, N):
...
+    for i in range(ny):
+        G[nx*i:nx*i+nx, 0] = i+1
+        G[nx*i:nx*i+nx, 1] = np.arange(1, nx+1)
+    m_est = np.dot(np.dot(lin.inv(np.dot(G.T, G)), G.T), d_obs)
+    d_est = np.dot(G, m_est)
+    lin_trend = np.reshape(d_est, (ny,nx))
+
+    da -= lin_trend
+
+    return da
```
Comment (rabernat) [inline xrft/xrft.py]: Does this function work with 1D data?

File: xrft/xrft.py [hunk 2]
```diff
@@ -95,8 +132,19 @@ def dft(da, dim=None, shift=True, remove_mean=True, window=False):
...
     k = [ np.fft.fftfreq(Nx, dx) for (Nx, dx) in zip(N, delta_x) ]
 
-    if remove_mean:
+    if detrend == 'constant':
         da = da - da.mean(dim=dim)
+    elif detrend == 'linear':
+        if len(axis_num) == 1:
+            da = xr.DataArray(sig.detrend(da.values, axis=axis_num[0],
+                                        type=detrend),
+                            dims=da.dims, coords=da.coords)
```
Comment (rabernat) [inline xrft/xrft.py]: Here you use the scipy function instead of your own function. Why?

How does scipy.signal.detrend handle dask arrays?

**Comment by rabernat** [comment]
This looks pretty good.

I'm just concerned that it might not play well with dask arrays.

File: xrft/xrft.py (hunk 1 above)
Comment (roxyboy) [inline xrft/xrft.py]: No, so I have a `if` syntax in the `dft` function to differentiate 1D and 2D.

File: xrft/xrft.py (hunk 2 above)
Comment (roxyboy) [inline xrft/xrft.py]: I played around with `scipy.signal.detrend` a bit but it didn't work when I tried to input more than one axis to detrend over...

**Comment by roxyboy** [comment]
Yea, you were write, `scipy.signal.detrend` doesn't keep the dask attribute

File: xrft/xrft.py (hunk 1 above)
Comment (rabernat) [inline xrft/xrft.py]: Try wrapping this with map_blocks

**Comment by roxyboy** [comment]
I think the `detrend` function works now for dask arrays but the output of `dask.array.map_blocks` returns a numpy array so if the data is detrended, the `fft` will be taken over an `xarray.DataArray` (for now).

**Comment by rabernat** [comment]
Ok this is fine for now.

We will have to clean up and refactor this a bit at some point over the summer.

**Pull request merged**